src/lib.rs

```diff
@@ -62,13 +65,15 @@
 #![no_std]
 #![cfg_attr(feature = "internal_benches", allow(unstable_features), feature(test))]
 
-#[cfg(any(test, feature = "use_heap"))]
+#[cfg(any(test, feature = "alloc"))]
+extern crate alloc;
+
+#[cfg(any(test, feature = "std"))]
```

Comment:
`#[cfg(feature = "std")]` is what's needed here, otherwise the current `cfg` make tests require `std`.